projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -103,10 +104,21 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
     @Input()
     canOpen = true;
 
+    private readonly hostHover$ = combineLatest([
+        tuiTypedFromEvent(this.el.nativeElement, 'mouseover'),
+        this.hover$ || EMPTY,
+    ]).pipe(
+        map(
+            ([{target}, hovered]) =>
+                hovered && this.computedHost.contains(target as HTMLElement),
```

Comment:
To take delays into account you can rewrite the first part of this stream as follows:
```ts
tuiTypedFromEvent(this.el.nativeElement, 'mouseover').pipe(
    map(
        e => this.computedHost.contains(tuiGetActualTarget(e)),
        switchMap(visible =>
            of(visible).pipe(
                delay(
                    (visible ? this.hover$?.showDelay : this.hover$?.hideDelay) || 0,
                ),
            ),
        ),
    ),
);
```

Reply:
fixed